frontend/react/src/pages/ViewCourse.jsx

```diff
@@ -0,0 +1,38 @@
+import React ,{useState, useEffect}from "react";
+import CourseView from "../components/view/CourseView";
+import {useParams} from "react-router-dom";
+
+export default function ViewCourse() {
+    let {id} = useParams();
+    id = parseInt(id);
+    const [isLoading, setIsLoading] = useState(true);
+    const [courseData, setCourseData] = useState(null);
+
+    //get request to get course details
+    useEffect(() => {
+        setIsLoading(true);
+        fetch(
```

Comment:
For consistency, it might be better to move this fetch into its own util function, like GetCourse(id) or something to match the other get functions. It would also be easier to mock for tests at the very least.